pubsub_nats.go

```diff
@@ -16,16 +16,35 @@ import (
 	"fmt"
 	"time"
 
-	"go.aporeto.io/elemental"
-
 	"github.com/gofrs/uuid"
-	nats "github.com/nats-io/go-nats"
+	"github.com/nats-io/go-nats"
+	"go.aporeto.io/elemental"
 	"go.uber.org/zap"
 )
 
+// NoClientError is returned in the event where a client was setup
+type NoClientError struct {
+	Message string
+}
+
+func (e *NoClientError) Error() string {
+	return e.Message
+}
+
+// EncodingError is returned in the event that the publication failed to be encoded
+// The underlying encoding error can be retrieved via 'Err'
+type EncodingError struct {
+	Message string
+	Err     error
+}
+
+func (e *EncodingError) Error() string {
+	return fmt.Sprintf("%s: %s", e.Message, e.Err)
+}
+
 type natsPubSub struct {
 	natsURL        string
-	client         *nats.Conn
+	client         NATSClient
```

Comment:
I don't think this interface needs to be public

Reply:
Done